### PR TITLE
add BitBTC token

### PR DESCRIPTION
**Pull request by iBitLabs**
**Description**
Added BitBTC L1 and L2 token details to tokens list
Added BitBTC logo to Logos folder

**Additional context**
Challenge to bridge BitBTC token to optimism

BitBTC can be obtained by staking WBTC(MockWBTC on kovan & kovan-optimistic) or other BTC-like tokens at a constant exchange rate of 1 million.

### CODE REVIEW

File: optimism.tokenlist.json [hunk 1]
```diff
@@ -637,6 +637,50 @@
...
+    },
+    {
+      "chainId": 1,
+      "address": "0x3C513dB8Bdc3806e4489d62C3d549A5Aaf6A4e97",
+      "name": "BitBTC",
+      "symbol": "BitBTC",
+      "decimals": 18,
+      "logoURI": "https://ethereum-optimism.github.io/logos/BitBTC.png",
+      "extensions": {
+        "optimismBridgeAddress": "0xaBA2c5F108F7E820C049D5Af70B16ac266c8f128"
```
Comment (elenadimitrova) [inline optimism.tokenlist.json:649]: What's the reason here for not using the standard bridge?

Comment (iBitLabs) [reply optimism.tokenlist.json:649]: To support L1 <=> L2 token mutual recognition, a custom bridge is required for BitBTC when new token was mined on L2 that should also be recognized by L1 according to [Optimism Document](https://community.optimism.io/docs/protocol/protocol-2.0.html#bridge-contracts). BitBTC(L2) was mined by l2Bridge when BitBTC(L1) is deposited to l1Bridge, and vice versa.

Comment (elenadimitrova) [reply optimism.tokenlist.json:649]: Sorry it's taking longer as I need to review the custom bridge. Why is this using an upgradability pattern but at the same time the ownership of it has been renounced i.e. the owner is set to 0x0?

File: optimism.tokenlist.json [hunk 2]
```diff
@@ -637,6 +637,50 @@
...
+    },
+    {
+      "chainId": 69,
+      "address": "0x83643C9eF0c5707A7815124754d0828c9A38be3a",
+      "name": "BitBTC",
+      "symbol": "BitBTC",
+      "decimals": 18,
+      "logoURI": "https://ethereum-optimism.github.io/logos/BitBTC.png",
+      "extensions": {
+        "optimismBridgeAddress": "0x0CFb46528a7002a7D8877a5F7a69b9AaF1A9058e"
```
Comment (elenadimitrova) [inline optimism.tokenlist.json:682]: Would you mind splitting the PR to allow us to test this on Kovan first

**Review by elenadimitrova — CHANGES_REQUESTED** [review]
I looked through the bridge implementation and see the following problems:
- the L1 bridge does not keep track of token pair balances instead it appears to just mint more tokens on L1 if the balance is insufficient
- owner, l2TokenBridge and messenger are not initialised on the L1 bridge and the _setOptimsimBridge function allows the owner to switch the L2 bridge. Also the L1 bridge is upgradeable
We'll need to be sure of the security you have implemented before we can add this to the token list and gateway. Happy to discuss in Discord or Telegram if you need further info.

File: optimism.tokenlist.json (hunk 1 above)
Comment (iBitLabs) [reply optimism.tokenlist.json:649]: 1. The BitBTC protocol will be deployed on EVM-based chains. Currently, L1 <-> L2 cross-chain protocol is very mature, BitBTC will be recognized each other between L1 and L2 via cross-chain bridge.
2. The custom bridge of BitBTC needs a mining function, and the mining function can only be called by the gateway. We can only set one gateway, if we set the gateway to a specific one, then we can no longer support other L2(Arbitrum etc.), so we can only use the upgradeable proxy contract to implement the gateway in a multiple inheritance manner.
3. BitBTC’s owner, l2TokenBridge, and messenger have been correctly set up, please [check](https://etherscan.io/address/0xaBA2c5F108F7E820C049D5Af70B16ac266c8f128#readProxyContract)

File: optimism.tokenlist.json (hunk 2 above)
Comment (iBitLabs) [reply optimism.tokenlist.json:682]: Okay, shall we resubmit a PR, or you help us? Thx.

Comment (elenadimitrova) [reply optimism.tokenlist.json:682]: Pls submit a PR with just the Kovan entries. Also pls answer these notes https://github.com/ethereum-optimism/ethereum-optimism.github.io/pull/45#pullrequestreview-825659719
Happy to chat on Discord or Telegram to move this faster if needed

**Comment by iBitLabs** [comment]
> I looked through the bridge implementation and see the following problems:
> 
> * the L1 bridge does not keep track of token pair balances instead it appears to just mint more tokens on L1 if the balance is insufficient. I assume this is to support withdrawals of WBTC on L2 to BitBTC on L1? But if that's the case why is the same amount burnt on L2 sent to L1 if the conversion rate is 1m to 1?
> * owner, l2TokenBridge and messenger are not initialised on the L1 bridge and the _setOptimsimBridge function allows the owner to switch the L2 bridge. Also the L1 bridge is upgradeable
>   We'll need to be sure of the security you have implemented before we can add this to the token list and gateway. Happy to discuss in Discord or Telegram if you need further info.

1. BitBTC and the L2 bridge happen to have a withdraw function with the same name. In fact, only BitBTC can cross-chain through the L1 bridge and the L2 bridge. Please let me explain this process again:

- L1: User call **BitBTC deposit** 0.0001 WBTC to the BitBTC contract on L1 and receives(**BitBTC _mint**) 100 BitBTC
- L1: User call **l1bridge depositERC20** 100 BitBTC on the L1 bridge. (Balance: +100 BitBTC)
- L2: **l2bridge brigdeMint** 100 BitBTC to user. (Balance: -100 BitBTC)
- L2: User call **BitBTC deposit** 0.0001 WBTC to the BitBTC contract on L2 and receives(**BitBTC _mint**) 100 BitBTC
- L2: User withdraws(**l2bridge withdraw**) 200 BitBTC back to L1. (Balance: +100 BitBTC)
- L1: 100 BitBTC were released(**ERC20 transfer**) and the left were mined(**l1bridge brigdeMint**) by the l1bridge. (Balance: -100 BitBTC)
- L1: User receives their 200 BitBTC on L1.

  The logic of withdraw of l2bridge is same as [the L2 standard bridge](https://optimistic.etherscan.io/address/0x4200000000000000000000000000000000000010#code).

2. BitBTC’s owner, l2TokenBridge, and messenger have been correctly set up, please [check](https://etherscan.io/address/0xaBA2c5F108F7E820C049D5Af70B16ac266c8f128#readProxyContract). The _setOptimsimBridge function can only be called once, because the second call will fail due to the _messenger is not address(0). The L1 bridge is upgradeable, this is the result we expect. In the future, the bridge will be upgraded to a new implementation. In addition to inheriting BitBTCL1BridgeOpImp, the new implementation will also inherit BitBTCL1BridgeArbImp.

3. Happy to discuss with you also.

File: optimism.tokenlist.json (hunk 2 above)
Comment (iBitLabs) [reply optimism.tokenlist.json:682]: We have resubmit A new [PR 46](https://github.com/ethereum-optimism/ethereum-optimism.github.io/pull/46).

**Comment by elenadimitrova** [comment]
Successfully tested [depositing](https://kovan.etherscan.io/tx/0x61a2a50bc346deb78074dec54b8387e1300927abcf832bbeeef3b83016cbf449) and [withdrawing](https://kovan-optimistic.etherscan.io/tx/0xfefa94adaebedd5a45983f6249faa49191b6b38c1937a97252a7175f91eafb4b) on kovan.

**Review by iBitLabs — COMMENTED** [review]
A [test PR](https://github.com/ethereum-optimism/ethereum-optimism.github.io/pull/46) has been submitted , and the test has passed.

**Comment by iBitLabs** [comment]
Requested changes have been merged. 
BitBTC: 
L1 -> L2 https://etherscan.io/tx/0x97ca8b18eb4ca9d153b73a4ad8cf1e4d328f5d9482a32a319d835a0f36e535fc
L2 -> L1 https://optimistic.etherscan.io/tx/0x3d55ab8ee51797081c03a4af68c3c5c1b0ef91b57232f9c0f7cd26ffb5a40141

**Review by elenadimitrova — APPROVED** [review]

**Pull request merged**